src/dbnode/storage/bootstrap.go

```diff
@@ -70,42 +70,154 @@ const (
+	i.log.Error("bootstrap prepare failed", zap.Error(err))
+}
+
+func (i *instrumentation) bootstrapStarted(shards int) {
+	i.logFields = []zapcore.Field{
+		zap.Int("numShards", shards),
+	}
+	i.log.Info("bootstrap started", i.logFields...)
+}
+
+func (i *instrumentation) bootstrapSucceeded() {
+	bootstrapDuration := i.nowFn().Sub(i.start)
+	i.bootstrapDuration.Record(bootstrapDuration)
+	i.logFields = append(i.logFields, zap.Duration("bootstrapDuration", bootstrapDuration))
+	i.log.Info("bootstrap succeeded, marking namespaces complete", i.logFields...)
+}
+
+func (i *instrumentation) bootstrapFailed(err error) {
+	bootstrapDuration := i.nowFn().Sub(i.start)
+	i.bootstrapDuration.Record(bootstrapDuration)
+	i.logFields = append(i.logFields, zap.Duration("bootstrapDuration", bootstrapDuration))
+	i.log.Error("bootstrap failed", append(i.logFields, zap.Error(err))...)
+}
+
+func (i *instrumentation) bootstrapNamespaceFailed(err error, namespaceID string) {
+	i.log.Info("bootstrap namespace error", append(i.logFields, []zapcore.Field{
+		zap.String("namespace", namespaceID),
+		zap.Error(err),
+	}...)...)
+}
+
+func (i *instrumentation) bootstrapNamespacesFailed(err error) {
+	duration := i.nowFn().Sub(i.startNamespaces)
+	i.bootstrapNamespacesDuration.Record(duration)
+	i.logFields = append(i.logFields, zap.Duration("bootstrapNamespacesDuration", duration))
+	i.log.Info("bootstrap namespaces failed", append(i.logFields, zap.Error(err))...)
+}
+
+func (i *instrumentation) bootstrapNamespacesStarted() {
+	i.startNamespaces = i.nowFn()
+	i.log.Info("bootstrap namespaces start", i.logFields...)
+}
+
+func (i *instrumentation) bootstrapNamespacesSucceeded() {
+	duration := i.nowFn().Sub(i.startNamespaces)
+	i.bootstrapNamespacesDuration.Record(duration)
+	i.logFields = append(i.logFields, zap.Duration("bootstrapNamespacesDuration", duration))
+	i.log.Info("bootstrap namespaces success", i.logFields...)
+}
+
+func (i *instrumentation) bootstrapCompletion() {
+	i.lastBootstrapCompletionTime = xtime.ToUnixNano(i.nowFn())
+}
+
+func (i *instrumentation) setIsBootstrapped(isBootstrapped bool) {
+	if isBootstrapped {
+		i.status.Update(1)
+	} else {
+		i.status.Update(0)
+	}
+}
+
+func (i *instrumentation) setIsBootstrappedAndDurable(isBootstrappedAndDurable bool) {
+	if isBootstrappedAndDurable {
+		i.durableStatus.Update(1)
+	} else {
+		i.durableStatus.Update(0)
+	}
+}
+
+func (i *instrumentation) missingNamespaceFromResult(err error) {
+	instrument.EmitAndLogInvariantViolation(i.opts.InstrumentOptions(), func(l *zap.Logger) {
+		l.Error("bootstrap failed", append(i.logFields, zap.Error(err))...)
+	})
+}
+
+func (i *instrumentation) bootstrapDataAccumulatorCloseFailed(err error) {
+	instrument.EmitAndLogInvariantViolation(i.opts.InstrumentOptions(),
+		func(l *zap.Logger) {
+			l.Error("could not close bootstrap data accumulator",
+				zap.Error(err))
+		})
+}
+
+func newInstrumentation(opts Options) *instrumentation {
```

Comment:
Nit: move `instrumentation` just after the `type instrumentation` and before its methods, I believe that's a more usual layout.